### DISCUSSION TITLE
Extract out reactive attributes

### DISCUSSION THREAD

**Discussion by davidbrochart** [Textualize/textual#2149] [Q&A] [answered]
I'm thinking about using Textual's [reactive attributes](https://textual.textualize.io/guide/reactivity) in another project. From early experiments, it seems that all they need is the class holding them to have a `_reactives` dictionary. I don't think the event system is mandatory, except maybe if watchers are async?
How would you feel about this, since it would basically be a copy/paste of Textual's code? If you agree with that, I guess the project should mention that the reactivity stuff comes from Textual.
What about the license? I guess it must be as permissive as Textual's, e.g. MIT of BSD.

**Comment by willmcgugan** [comment, marked as answer]
I'm happy for you to do that. 

Not sure what the MIT license says about derived works, but I'd have no issue with licensing it as MIT or BSD.

**Reply by davidbrochart** [reply]
Thanks, I created [reacttrs](https://github.com/davidbrochart/reacttrs).

**Reply by willmcgugan** [reply]
Clever name!
